connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1403,6 +1403,9 @@ class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
     if (attr.hasPlanId) {
       expr.setTagValue(LogicalPlan.PLAN_ID_TAG, attr.getPlanId)
     }
+    if (attr.hasIsMetadataColumn) {
```

Comment:
For the record I am not sure I like this. It gets the job done, but I do feel we should use actual fields for this instead of sneaking it in.